flutter_local_notifications/example/lib/main.dart

```diff
@@ -310,6 +310,29 @@ class _HomePageState extends State<HomePage> {
     }
   }
 
+  Future<void> _requestProvisionalPermissions() async {
```

Comment:
How does this work within the context of the example app? From what I could tell, it doesn't align with current experience as the example app requests the permissions on startup. Once you give or don't give permissions at the prompt, the prompt no longer appears and requires going to app settings to change and calling this doesn't actually have an effect. Moreover, most users will grant permissions on startup. With that in mind, unless I have missed anything (do let me know if I have) then this looks like this would be something better left out of the example app. Don't want the example app to default to request provisional permissions on startup instead either as it's likely rare for apps to use this and could leave to more questions being raised around features not working properly

Reply:
You are correct. I will revert the changes to leave the file untouched.

When testing the provisional feature, we should follow these steps:

1. Comment out `requestPermission`.
2. Clear the permission in the system's settings -> notifications if it was previously installed.
3. Call `requestProvisionalPermission`.
If the user had previously rejected the permission before we called `requestProvisionalPermission`, then `requestProvisionalPermission` will not work until step 2 is completed.


Addressed at [77a2d07](https://github.com/MaikuB/flutter_local_notifications/pull/2022/commits/77a2d074bf80e3ac6b864f094f28eada82aa5e10)